bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -0,0 +1,307 @@
+                    continue;
+                }
+
+                State state = null;
+
+                switch (dataItem.getDataType()) {
+                    case Flags:
+                        state = TypeConverter.toOnOffType(message.getBit(dataItem.getByteType(), dataItem.getBitPos()));
+                        break;
+                    case Uint8:
+                    case Uint16:
+                        state = TypeConverter.toDecimalType(message.getUInt(dataItem.getByteType()));
+                        break;
+                    case Int8:
+                    case Int16:
+                        state = TypeConverter.toDecimalType(message.getInt(dataItem.getByteType()));
+                        break;
+                    case Float:
+                        state = TypeConverter.toDecimalType(message.getFloat());
+                        break;
+                    case DoWToD:
+                        break;
+                }
+
+                if (state != null) {
+                    logger.debug("Received update for channel '{}': {}", channelId, state.toFullString());
+                    try {
+                        updateState(channelId, state);
+                    } catch (IllegalStateException e) {
+                        // Missing callback, possibly due to incorrect initialization. how to handle correctly ?
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void log(LogLevel loglevel, String format, String arg) {
+        String message = String.format(format, arg);
+        log(loglevel, message);
+    }
+
+    @Override
+    public void log(LogLevel loglevel, String message, Throwable t) {
+        switch (loglevel) {
+            case Trace:
+                logger.trace("{}", message, t);
+                break;
+            case Debug:
+                logger.debug("{}", message, t);
+                break;
+            case Info:
+                logger.info("{}", message, t);
+                break;
+            case Warning:
+                logger.warn("{}", message, t);
+                break;
+            case Error:
+                logger.error("{}", message, t);
+                break;
+            default:
+                break;
+        }
+    }
+
+    @Override
+    public void log(LogLevel loglevel, String message) {
+        switch (loglevel) {
+            case Trace:
+                logger.trace("{}", message);
+                break;
+            case Debug:
+                logger.debug("{}", message);
+                break;
+            case Info:
+                logger.info("{}", message);
+                break;
+            case Warning:
+                logger.warn("{}", message);
+                break;
+            case Error:
+                logger.error("{}", message);
+                break;
+            default:
+                break;
+        }
+    }
```

Comment:
Please remove this. Don't add custom logging features.

Reply:
The thought behind this was that the connector running in a separate thread didn't have access to the logging framework. Therefor a mechanism was added that logs via a callback to the thinghandler, which in turns logs to logging framework. Will test to see if I can log from the thread, then this part of the callback interface can be removed.

Reply:
After running some tests, removed logging callback and let connector log directly to logging framework